include/simdjson/generic/ondemand/document.h

```diff
@@ -263,6 +263,17 @@ class document {
   /** @overload simdjson_really_inline simdjson_result<value> find_field_unordered(std::string_view key) & noexcept; */
   simdjson_really_inline simdjson_result<value> operator[](const char *key) & noexcept;
 
+  /**
+   * Get the type of this JSON value.
```

Comment:
I would list the possible values: json_type::array,  json_type::object, json_type::number, json_type::string,  json_type::boolean, json_type::null.

(Optional comment.)

Reply:
That's a good idea here. I'll do that.